californium-core/src/test/java/org/eclipse/californium/core/test/ClientSynchronousTest.java

```diff
@@ -150,6 +153,34 @@ public void testSynchronousCall() throws Exception {
 		Assert.assertEquals(5, notifications.get());
 		Assert.assertFalse(failed);
 	}
+
+	@Test
+	public void testAdvancedUsesTypeFromRequest() throws Exception {
+		String uri = "coap://localhost:"+serverPort+"/"+TARGET;
+		CoapClient client = new CoapClient(uri).useExecutor();
+
+		// Set NONs but expecting CONs as specified in request
+		client.useNONs();
+
+		Request request = new Request(Code.GET, Type.CON);
+
+		CoapResponse resp = client.advanced(request);
+
+		Assert.assertEquals(CONTENT_1, resp.getResponseText());
```

Comment:
How does this assert that the request indeed contained a CON? The fact that we receive a response merely means that the request has been received and processed by the server ... We should at least assert that the response is of type ACK, i.e. the server piggybacked the response, shouldn't we?